hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -1841,15 +1848,15 @@ case class TableAggregateByKey(child: TableIR, expr: IR) extends TableIR {
 
   protected[ir] override def execute(ctx: ExecuteContext): TableValue = {
     val prev = child.execute(ctx)
-    val prevRVD = prev.rvd
+    val prevRVD = prev.rvd.truncateKey(child.typ.key.length)
```

Comment:
I think it's simpler and clearer to use `prev.rvd.truncateKey(child.typ.key)`, which makes the assertion below redundant.

Reply:
oh, good call. didn't find that method.